### PR TITLE
CORTX-30473: add support bundle filters to logs-cortx-cloud.sh

### PR DESCRIPTION
**Pull request by rohit-k-dwivedi**
Signed-off-by: Rohit Dwivedi <rohit.k.dwivedi@seagate.com>
JIRA: https://jts.seagate.com/browse/CORTX-30473

## Description
Add all support bundle filter options to log-cortx-cloud.sh

## Breaking change
No breaking changes, the feature is backward compatible

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds new functionality)
- [ ] Breaking change (bug fix or new feature that breaks existing functionality)
- [ ] Third-party dependency update
- [ ] Documentation additions or improvements
- [ ] Code quality improvements to existing code or test additions/updates

## Applicable issues
NA

- This change fixes an issue: #
- This change is related to an issue: #

## CORTX image version requirements
NA

This change requires the following images:

- `cortx-all:<version>`
- `cortx-rgw:<version>`

## How was this tested?
deployed the setup with current changes and triggered the log-cortx-cloud.sh with and without filter

## Checklist

- [x] The change is tested and works locally.
- [ ] New or changed settings in the solution YAML are documented clearly in the README.md file.
- [x] All commits are signed off and are in agreement with the [CORTX Community DCO and CLA policy](https://github.com/Seagate/cortx/blob/main/doc/dco_cla.md).

If this change addresses a CORTX Jira issue:

- [x] The title of the PR starts with the issue ID (e.g. `CORTX-XXXXX:`)

### CODE REVIEW

**Comment by rahul-tripathi-git** [comment]
Hi @rohit-k-dwivedi 
Code-wise everything looks good, can you please verify the behavior of SB with filters.
CC: @cdeshmukh

File: k8_cortx_cloud/logs-cortx-cloud.sh [hunk 1]
```diff
@@ -23,6 +23,13 @@ date=$(date +%F_%H-%M)
 solution_yaml=${1:-"solution.yaml"}
 nodename=""
 pods_found=0
+size_limit="500MB"
+duration="P5D"
+binlogs="False"
+coredumps="False"
+stacktrace="False"
+all="True"
```
Comment (cdeshmukh) [inline k8_cortx_cloud/logs-cortx-cloud.sh:31]: default for 'all' should be 'False' only

File: k8_cortx_cloud/logs-cortx-cloud.sh [hunk 2]
```diff
@@ -23,6 +23,13 @@ date=$(date +%F_%H-%M)
 solution_yaml=${1:-"solution.yaml"}
 nodename=""
 pods_found=0
+size_limit="500MB"
```
Comment (cdeshmukh) [inline k8_cortx_cloud/logs-cortx-cloud.sh:26]: Need to update usage() also with these new optional params

Comment (rohit-k-dwivedi) [reply k8_cortx_cloud/logs-cortx-cloud.sh:43]: done

File: k8_cortx_cloud/logs-cortx-cloud.sh (hunk 1 above)
Comment (rohit-k-dwivedi) [reply k8_cortx_cloud/logs-cortx-cloud.sh]: done

File: k8_cortx_cloud/logs-cortx-cloud.sh [hunk 3]
```diff
@@ -16,13 +16,26 @@ function usage() {
   echo "                                     default file path is ${solution_yaml}."
   echo "    -n|--nodename NODENAME: collects logs from pods running only on given node".
   echo "                            collects logs from all the nodes by default."
+  echo "    --duration DURATION : dutaion for which logs should be collected"
+  echo "    --size_limit SIZE : max log size "
```
Comment (rahul-tripathi-git) [inline k8_cortx_cloud/logs-cortx-cloud.sh:20]: max size limit for support bundle to be generated. Or something like this.

File: k8_cortx_cloud/logs-cortx-cloud.sh [hunk 4]
```diff
@@ -16,13 +16,26 @@ function usage() {
...
+  echo "    --binlogs True/False : option to collect binary logs"
+  echo "    --coredumps True/False : option to collect core dumps"
+  echo "    --stacktrace True/False : option to collect stack trace"
+  echo "    --all True/False : if True, sets --binlogs --coredumps & --stacktrace to True"
```
Comment (rahul-tripathi-git) [inline k8_cortx_cloud/logs-cortx-cloud.sh:24]: aggregated option to collect binlogs, coredumps & stacktrace.
If set to true, It overrides --binlogs, --coredumps & --stacktrace at once.

**Review by rahul-tripathi-git — CHANGES_REQUESTED** [review]
Put a full stop on each usage lines added.

File: k8_cortx_cloud/logs-cortx-cloud.sh (hunk 2 above)
Comment (cdeshmukh) [reply k8_cortx_cloud/logs-cortx-cloud.sh:32]: Ok. Am not getting option to resolve the

**Review by cdeshmukh — APPROVED** [review]
Am ok after addressing comments raised by @rahul-tripathi-git

File: k8_cortx_cloud/logs-cortx-cloud.sh (hunk 4 above)
Comment (rohit-k-dwivedi) [reply k8_cortx_cloud/logs-cortx-cloud.sh]: done

File: k8_cortx_cloud/logs-cortx-cloud.sh (hunk 3 above)
Comment (rohit-k-dwivedi) [reply k8_cortx_cloud/logs-cortx-cloud.sh]: done

File: k8_cortx_cloud/logs-cortx-cloud.sh (hunk 2 above)
Comment (rohit-k-dwivedi) [reply k8_cortx_cloud/logs-cortx-cloud.sh:34]: I see that option, resolving the comment

File: k8_cortx_cloud/logs-cortx-cloud.sh [hunk 5]
```diff
@@ -16,13 +16,28 @@ function usage() {
   echo "                                     default file path is ${solution_yaml}."
   echo "    -n|--nodename NODENAME: collects logs from pods running only on given node".
   echo "                            collects logs from all the nodes by default."
+  echo "    --duration DURATION : dutaion for which logs should be collected"
```
Comment (ujjwalpl) [inline k8_cortx_cloud/logs-cortx-cloud.sh:19]: These options should be added to the usage command line. 
```
echo -e "Usage: \`sh $0 [<optons>]\`\n"
Where options are:
```
Also, please add short options for all parameters.

Comment (ujjwalpl) [inline k8_cortx_cloud/logs-cortx-cloud.sh:19]: These options should be added to the usage command line. 
```
echo -e "Usage: \`sh $0 [<optons>]\`\n"
Where options are:
```
Also, please add short options for all parameters.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

**Comment by rohit-k-dwivedi** [comment]
@walterlopatka , Not every component has implemented all the filters, only selected filters are currently implemented,  hence for now there might not be much difference in support bundles with filters, 

1. https://github.com/Seagate/cortx-ha/blob/1d9af2a1395914b79b0d7e203b59e06aedbc3374/ha/core/support_bundle/ha_support_bundle.py#L36
2. https://github.com/Seagate/cortx-rgw-integration/blob/4347893e7814dea4a8202815dde90aac887c47e6/src/rgw/support/rgw_support_bundle#L53

**Review by keithpine — APPROVED** [review]

File: k8_cortx_cloud/logs-cortx-cloud.sh [hunk 6]
```diff
@@ -103,7 +132,14 @@ function getInnerLogs()
       --cluster_conf_path yaml:///etc/cortx/cluster.conf \
       --location file://${path} \
       --bundle_id "${name}" \
-      --message "${name}"
+      --message "${name}" \
```
Comment (keithpine) [inline k8_cortx_cloud/logs-cortx-cloud.sh:135]: Just noticed this is set to an empty string. Is that correct?

Comment (cdeshmukh) [reply k8_cortx_cloud/logs-cortx-cloud.sh:135]: what exactly set to empty? @rohit-k-dwivedi plz follow up and fix if needed for anything

**Pull request merged**